### PR TITLE
[COR-545] validate sqs body

### PR DESCRIPTION
**Pull request by morus12**

### CODE REVIEW

**Review by nmarie22 — APPROVED** [review]

File: messagequeue/sqs/validation.go [hunk 1]
```diff
@@ -82,9 +96,15 @@ func BodyIsValid(s string) error {
 	if utf8.RuneCountInString(s) == 0 {
 		return errors.New(bodyMinimumError)
 	}
-	if len(s) > maxBodyKilobytes {
-		return errors.New("body cannot exceed %d kilobytes (was %d kb)",
-			maxBodyKilobytes, len(s)/1024)
+
+	if len(s) > maxBodyKilobytes*1000 {
```
Comment (apynes) [inline messagequeue/sqs/validation.go:100]:
```suggestion
	if len(s) > maxBodyKibibytes*1024 {
```

File: messagequeue/sqs/validation.go [hunk 2]
```diff
@@ -13,7 +14,7 @@ const (
 	minNameCharacters = 1
 	maxNameCharacters = 256
 	minBodyCharacters = 1
-	maxBodyKilobytes  = 255
+	maxBodyKilobytes  = 256
```
Comment (apynes) [inline messagequeue/sqs/validation.go:17]:
```suggestion
	maxBodyKibibytes  = 256
```

File: messagequeue/sqs/validation.go [hunk 3]
```diff
@@ -82,9 +96,15 @@ func BodyIsValid(s string) error {
 	if utf8.RuneCountInString(s) == 0 {
 		return errors.New(bodyMinimumError)
 	}
-	if len(s) > maxBodyKilobytes {
-		return errors.New("body cannot exceed %d kilobytes (was %d kb)",
-			maxBodyKilobytes, len(s)/1024)
+
+	if len(s) > maxBodyKilobytes*1000 {
+		return errors.New("body cannot exceed %d kilobytes (was %d bytes)", maxBodyKilobytes, len(s))
```
Comment (apynes) [inline messagequeue/sqs/validation.go:101]:
```suggestion
		return errors.New("body cannot exceed %d KiB (was %d bytes)", maxBodyKibibytes, len(s))
```

File: messagequeue/sqs/validation.go [hunk 4]
```diff
@@ -82,9 +96,15 @@ func BodyIsValid(s string) error {
...
+	}
+
+	for _, r := range s {
+		if !unicode.In(r, allowedRanges) {
+			return errors.New("char 0x%x is not allowed unicode character", r)
```
Comment (apynes) [inline messagequeue/sqs/validation.go:106]:
```suggestion
			return errors.New("body cannot contain forbidden unicode character 0x%x", r)
```

**Review by apynes — CHANGES_REQUESTED** [review]
disambiguation of Kibi and Kilo. The requirements are for 256 Kibibytes (JEDEC Kilobytes) not Kilobytes.
Also, we can clean up a error message to be more clear.

**Review by apynes — APPROVED** [review]

**Pull request merged**